pkg/storage/cloud/cloud_io.go

```diff
@@ -190,13 +191,17 @@ func (r *ResumingReader) Read(p []byte) (int, error) {
 			log.Errorf(r.Ctx, "Read err: %s", lastErr)
 		}
 
-		if isResumableHTTPError(lastErr) {
-			if retries >= maxNoProgressReads {
-				return 0, errors.Wrap(lastErr, "multiple Read calls return no data")
+		// Check if the ResumingReader has been configured with retry-on-error
+		// decider.
+		if r.RetryOnErrFn != nil {
```

Comment:
I wonder if we should log if RetryOnErrFn is nil (so we're not retrying) and/or fall back to a default function (e.g. sysutil.IsErrConnectionReset) ?

Reply:
Makes sense, I moved the struct creation to a NewResumingReader method to centralize the init.